keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/ActiveServerGroupImageDeserializer.kt

```diff
@@ -8,14 +8,17 @@ import com.netflix.spinnaker.keel.api.ec2.ServerGroup.ActiveServerGroupImage
 import org.springframework.boot.jackson.JsonComponent
 
 @JsonComponent
-class ActiveServerGroupImageDeserializer : StdNodeBasedDeserializer<ActiveServerGroupImage>(ActiveServerGroupImage::class.java) {
+class ActiveServerGroupImageDeserializer :
+  StdNodeBasedDeserializer<ActiveServerGroupImage>(ActiveServerGroupImage::class.java) {
   override fun convert(root: JsonNode, ctxt: DeserializationContext): ActiveServerGroupImage {
     val tags = root.get("tags") as ArrayNode
 
     return ActiveServerGroupImage(
       imageId = root.get("imageId").textValue(),
       appVersion = tags.getTag("appversion")?.substringBefore("/"),
-      baseImageVersion = tags.getTag("base_ami_version"),
+      baseImageName = root.get("description").textValue()?.let {
+        Regex("""ancestor_name=([\w-]+)""").find(it)?.groupValues?.get(1)
```

Comment:
Givent that the identical parsing logic is in the ActiveServerGroup.kt class, (and that this involves a regex), I'd recommend encapsulating this logic in a function.

Reply:
I'm inclined to agree

Reply:
Done